nebula-logger/plugins/log-retention-rules/plugin/classes/LogRetentionFilter.cls

```diff
@@ -0,0 +1,464 @@
+
+            if (this.rule.ConditionLogicType__c != 'Custom') {
+                List<String> standardLogicPieces = new List<String>();
+                for (Integer i = 0; i < this.ruleFilterConditions.size(); i++) {
+                    standardLogicPieces.add(String.valueOf(i + 1));
+                }
+                conditionsLogic = '(' + String.join(standardLogicPieces, ' ' + this.rule.ConditionLogicType__c + ' ') + ')';
+            }
+
+            List<String> parsedCharacters = new List<String>();
+            Boolean hasFoundNumbers = false;
+            String foundNumberString = '';
+
+            for (String character : conditionsLogic.split('')) {
+                if (!character.isNumeric() && !hasFoundNumbers) {
+                    parsedCharacters.add(character);
+                } else if (!character.isNumeric() && hasFoundNumbers) {
+                    hasFoundNumbers = false;
+                    Integer foundNumber = Integer.valueOf(foundNumberString) - 1;
+
+                    parsedCharacters.add('{' + foundNumber + '}');
+                    foundNumberString = '';
+                    parsedCharacters.add(character);
+                } else if (character.isNumeric()) {
+                    hasFoundNumbers = true;
+                    foundNumberString += character;
+                } else if (hasFoundNumbers && !character.isNumeric() && !String.isBlank(foundNumberString)) {
+                    Integer foundNumber = Integer.valueOf(foundNumberString) - 1;
+
+                    parsedCharacters.add('{' + foundNumber + '}');
+                    foundNumberString = '';
+                } else {
+                    parsedCharacters.add(character);
+                }
+            }
+            return String.join(parsedCharacters, '').toUpperCase();
+        }
+    }
+
+    @SuppressWarnings('PMD.ApexDoc')
+    @TestVisible
+    private class FilterConditionResult {
+        private Boolean matchesFilter;
+        private SObject record;
+        private LogRetentionRuleCondition__mdt filterCondition;
+        private Schema.SObjectType sobjectType;
+
+        private FilterConditionResult(SObject record, LogRetentionRuleCondition__mdt filterCondition) {
+            this.sobjectType = record.getSObjectType();
+            this.record = record;
+            this.filterCondition = filterCondition;
+
+            this.matchesFilter = this.matchesFilter();
+        }
+
+        public String getCondition() {
+            return this.filterCondition.FieldPath__c +
+                ' ' +
+                this.filterCondition.Operator__c +
+                ' ' +
+                this.getComparisonValue() +
+                ' (' +
+                this.filterCondition.ValueType__c +
+                ')';
+        }
+
+        public Boolean matchesFilter() {
+            if (this.filterCondition.ValueType__c == 'RegEx') {
```

Comment:
long term, I feel like these strings could be represented by a private enum that you always fetch when referencing `ValueType__c`, and maybe short term have constants for them? This is really a nice to have, but just something I noticed while scrolling through this file

Reply:
Yeah, I love that idea - I'll add a `TODO` to make that change in a future release.